napari/plugins/pypi.py

```diff
@@ -0,0 +1,63 @@
+"""
+These convenience functions will be useful for searching pypi for packages
+that match the plugin naming convention, and retrieving related metadata.
+"""
+import requests
```

Comment:
Does pluggy pull in the requests dependency? Should we add `requests` to our dependencies because of this file? I feel like anything that we explicitly import should go in our dependencies.

Reply:
not pluggy.  usually it's all over the dependencies, but it looks like only `numpydoc` depends on it for us.  I can either explicitly add it to our dependencies, or not use it at all and revert to stdlib functions.  I prefer requests (and most people will already have it in their env due to its ubiquity) but your call.

Reply:
i added it. so @jni, just let me know if you prefer `urllib`

Reply:
removed the requests dependency